x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/CacheKey.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.searchablesnapshots.cache;
+
+import org.elasticsearch.index.shard.ShardId;
+import org.elasticsearch.repositories.IndexId;
+import org.elasticsearch.snapshots.SnapshotId;
+
+import java.nio.file.Path;
+import java.util.Objects;
+
+public class CacheKey {
+
+    private final SnapshotId snapshotId;
+    private final IndexId indexId;
+    private final ShardId shardId;
+    private final Path cacheDir;
```

Comment:
Similarly `cacheDir`. In my mind the thing in the cache is the Lucene file, which is uniquely identified by snapshot/index/shard/filename, and `cacheDir` is a bit of an implementation detail that I feel should be hidden here.

Reply:
Same here, thanks